hibernate-core/src/main/java/org/hibernate/loader/Loader.java

```diff
@@ -1184,22 +1193,40 @@ private void initializeEntitiesAndCollections(
 			post = null;
 		}
 
-		if ( hydratedObjects != null ) {
-			int hydratedObjectsSize = hydratedObjects.size();
-			LOG.tracev( "Total objects hydrated: {0}", hydratedObjectsSize );
+		if ( nullSeparatedHydratedObjects != null && !nullSeparatedHydratedObjects.isEmpty() ) {
+			if ( LOG.isTraceEnabled() ) {
+				int hydratedObjectsSize = 0;
+				for ( Object hydratedObject : nullSeparatedHydratedObjects ) {
+					if ( hydratedObject != null ) {
+						++hydratedObjectsSize;
+					}
+				}
+				LOG.tracev( "Total objects hydrated: {0}", hydratedObjectsSize );
```

Comment:
Should the first format string to be consistent with the one of `String#format()`? Then we should use `%d` as opposed to `{0}`?